### ISSUE TITLE
Study changes to t/op/*.t (3 of 12)

### ISSUE THREAD

**Issue by jkeenan** [atoomic/perl#213]
In this ticket we will study our progress in pursuit of [Objective 2, "strict-by-default"](https://github.com/atoomic/perl/issues/153), for the following files in the Perl core distribution:

```
t/op/const-optree.t
t/op/context.t
t/op/coreamp.t
t/op/coresubs.t
t/op/cproto.t
t/op/current_sub.t
t/op/dbm.t
t/op/decl-refs.t
t/op/defined.t
t/op/defins.t
```
These files should be evaluated in light of the guidelines presented on [this wiki page](https://github.com/atoomic/perl/wiki/How-to-Start-Working-with-Tests-in-the-Perl-Core-Distribution).  Please post the summary of your findings in a single comment in this issue.

**First-time contributors:**  (1) Please post a comment to this ticket indicating you are working on it.  That will enable us to assign the ticket to you.  (GitHub won't let us do that until it already thinks you're part of the project.)  (2) You should subscribe to this repository so that you get email updates.

Thank you for helping to improve the Perl programming language.

**Comment by oodler577** [comment]
@jkeenan interested in taking this

**Comment by oodler577** [comment]
t/op/const-optree.t
* looks good*

t/op/context.t
* looks good

t/op/coreamp.t
* looks ok
* uses "no strict 'refs';" in places, maybe able to be rectified
* uses "no strict 'subs';" in places, maybe able to be rectified
* out of scope changes seem to be included, e.g.,:

```
         eval " &CORE::$o(1) ";
    -    like $@, qr/^Too many arguments for $o at /, "&$o with too many args";
    +    like $@, qr/^Too many arguments for $o at /, "&$o with too many args ; line:" . __LINE__;
```

t/op/coresubs.t
* looks good
* uses "no strict 'refs';", can probably be rectified

t/op/cproto.t
* looks good

t/op/current_sub.t
* looks good
* uses "no strict 'subs'", maybe worth an attempt to rectify

t/op/dbm.t
* looks good

t/op/decl-refs.t
* looks good
* eval's code that contains "no strict 'vars'", which can probably be rectified easily

t/op/defined.t
* looks good
* uses "no warnings 'void';", which can probably be rectified easily

t/op/defins.t
* looks good
* uses scoped "no strict" in some places

**Comment by oodler577** [comment]
back to you @jkeenan

**Comment by jkeenan** [comment]
> t/op/const-optree.t
> 
>     * looks good*
> 
> 
> t/op/context.t
> 
>     * looks good
> 
> 
> t/op/coreamp.t
> 
>     * looks ok
> 
>     * uses "no strict 'refs';" in places, maybe able to be rectified
> 
>     * uses "no strict 'subs';" in places, maybe able to be rectified
> 

I'm unclear as to what you mean by "may be able to be rectified"; can you clarify?

Thank you very much.
Jim Keenan

**Comment by oodler577** [comment]
I am only indicating that it might be worth an attempt to remove localized weakening of strict in those cases. I am happy to give it a shot where ever it appears in the #213, #214, and #215

**Comment by jkeenan** [comment]
> I am only indicating that it might be worth an attempt to remove localized weakening of strict in those cases. I am happy to give it a shot where ever it appears in the #213, #214, and #215

Well, if we (in particular, I) weakened strictures in the course of work on a file since we forked from `v5.32.0`, then we presumably did it for a reason.  So, if you were to comment out a statement like `no strict 'refs';` in a scoped block and the test subsequently failed -- that would have been the reason.

Here you might want to use combinations of `git log`, `git blame` and `git show` to see what changed in a particular commit.

Based on your reports, I've closed #214 and #215, so you can confine your scope to this issue.

Thank you very much.
Jim Keenan

**Comment by oodler577** [comment]
Thanks fine with me. I can refrain from making that comment unless I have a true reason to suggest it, like it might not be necessary at all. I saw no cases here, but have seen 1 or 2 in the past reviews. I am fine with you closing this issue. Thanks!
